flake.nix

```diff
@@ -0,0 +1,85 @@
+{
+  description = "go-centrifuge";
+
+  inputs = {
+    nixpkgs.url = github:NixOS/nixpkgs/nixos-21.11;
+    gitignore = {
+      url = github:hercules-ci/gitignore.nix;
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs = { self, nixpkgs, gitignore }:
+    let
+      name = "go-centrifuge";
+      version = "2.0";
+
+      system = "x86_64-linux";
+      pkgs = nixpkgs.legacyPackages.${system};
+
+      srcFilter = src:
+        let
+          srcIgnored = gitignore.lib.gitignoreFilter src;
+          ignoreList = [
+            ".envrc"
+            ".github"
+            "CODE_OF_CONDUCT.md"
+            "README.md"
+            "codecov.yml"
+            "flake.lock"
+            "flake.nix"
+          ];
+        in
+        path: type:
+          srcIgnored path type
+          && builtins.all (name: builtins.baseNameOf path != name) ignoreList;
+    in
+    {
+      packages.${system} = {
+        go-centrifuge = pkgs.buildGoModule {
```

Comment:
Sort of same question as in the other PR, how to know the go version is used, probably part of the same presentation we have to shedule.

Reply:
The most basic way to know that is to:

- inspect which commit of nixpkgs we're using in this repo, which is found in the `flake.lock`
- Run `nix run nixpkgs/$commit#go -- version`

In this specific case:

```
❯ nix run nixpkgs/6f07605b6916abee5435ea1145a7541299a87c1b#go -- version
go version go1.16.9 linux/amd64
```

But I've also implemented a simpler solution, which works by dropping you in a shell where Go is installed. Since nixpkgs is pinned, this is the same Go used to build the app.

- `nix develop`
- `go version`